packages/tests/server/react/getQueryKey.test.tsx

```diff
@@ -39,6 +58,226 @@ const ctx = konn()
   .done();
 
 describe('getQueryKeys', () => {
+  test('no input', async () => {
+    const { proxy, App } = ctx;
+
+    function MyComponent() {
+      const happy1 = getQueryKey(proxy.post.all, undefined, 'query');
+      const happy2 = getQueryKey(proxy.post.all);
+
+      // @ts-expect-error - post.all has no input
+      const sad = getQueryKey(proxy.post.all, 'foo');
+
+      return (
+        <>
+          <pre data-testid="qKey1">{JSON.stringify(happy1)}</pre>
+          <pre data-testid="qKey2">{JSON.stringify(happy2)}</pre>
+        </>
+      );
+    }
+
+    const utils = render(
+      <App>
+        <MyComponent />
+      </App>,
+    );
+
+    await waitFor(() => {
+      expect(utils.getByTestId('qKey1')).toHaveTextContent(
+        JSON.stringify([['post', 'all'], { type: 'query' }]),
+      );
+      expect(utils.getByTestId('qKey2')).toHaveTextContent(
+        JSON.stringify([['post', 'all']]),
+      );
+    });
+  });
+
+  test('with input', async () => {
+    const { proxy, App } = ctx;
+
+    function MyComponent() {
+      const happy1 = getQueryKey(proxy.post.byId, { id: 1 }, 'query');
+
+      // doesn't really make sense but should still work
+      const happyIsh = getQueryKey(proxy.post.byId, { id: 1 });
+
+      // @ts-expect-error - post.byId has required input
+      const sad = getQueryKey(proxy.post.byId, undefined, 'query');
+
+      return (
+        <>
+          <pre data-testid="qKey1">{JSON.stringify(happy1)}</pre>
+          <pre data-testid="qKey2">{JSON.stringify(happyIsh)}</pre>
+        </>
+      );
+    }
+
+    const utils = render(
+      <App>
+        <MyComponent />
+      </App>,
+    );
+
+    await waitFor(() => {
+      expect(utils.getByTestId('qKey1')).toHaveTextContent(
+        JSON.stringify([['post', 'byId'], { input: { id: 1 }, type: 'query' }]),
+      );
+      expect(utils.getByTestId('qKey2')).toHaveTextContent(
+        JSON.stringify([['post', 'byId'], { input: { id: 1 } }]),
+      );
+    });
+  });
+
+  test('undefined input but type', async () => {
+    const { proxy, App } = ctx;
+
+    function MyComponent() {
+      const happy = getQueryKey(proxy.post.all, undefined, 'query');
+
+      return (
+        <>
+          <pre data-testid="qKey">{JSON.stringify(happy)}</pre>
+        </>
+      );
+    }
+
+    const utils = render(
+      <App>
+        <MyComponent />
+      </App>,
+    );
+
+    await waitFor(() => {
+      expect(utils.getByTestId('qKey')).toHaveTextContent(
+        JSON.stringify([['post', 'list'], { type: 'query' }]),
+      );
+    });
+  });
+
+  test('infinite', async () => {
+    const { proxy, App } = ctx;
+
+    function MyComponent() {
+      const happy = getQueryKey(proxy.post.list, undefined, 'infinite');
```

Comment:
i think this should work, or is the cursor part of the queryKey and mandatory - honestly not sure

Reply:
Yep, just verified. The cursor is not part of the queryKey so this should indeed be fine: 
<img width="503" alt="CleanShot 2023-01-22 at 20 07 33@2x" src="https://user-images.githubusercontent.com/51714798/213934960-8799cd93-8b87-4122-b382-bfb23c065279.png">